components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/file/FileInjectHandler.java

```diff
@@ -175,6 +176,11 @@ public boolean invoke(Object object, String name)throws SynapseException{
          if(dataSource != null) {
 				dataSource.destroy();
 			}
+            try {
+                in.close();
```

Comment:
Can we use try with resources instead?

Reply:
We can't use try with resource in this case as it can't be a final.